tsinfer/formats.py

```diff
@@ -634,11 +644,15 @@ class Site(object):
 class Variant(object):
     """
     A single variant. Mirrors the definition in tskit but with some extra fields.
+    The inference_time field returns the time used for inference: where a site
+    is using frequency as a proxy for time, this will contain the frequency, otherwise
+    it will simply contain the site.time value.
     """
 
     # TODO document properly.
     site = attr.ib()
     genotypes = attr.ib()
+    inference_time = attr.ib()
```

Comment:
This is a property of the Site isn't it?

Reply:
This has to be a property of the Variant, because it it is a function of the genotypes (which the Site doesn't have access to). The Site only has access to the passed-in time, which (when time==freq) is the marker value `-inf`. We can't add it to the Site without slowing down the Sites iterator by grabbing the genotypes on iteration.